test/Test/Data/Mutable/Set.hs

```diff
@@ -2,18 +2,56 @@
 {-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# OPTIONS_GHC -Wno-name-shadowing #-}
-
 -- |
 -- Tests for mutable sets.
-
+--
+-- = How we designed the tests
+--
+-- We use hedgehog to test properties that are axioms that fully define the
+-- behavior of sets. There is at least one axiom per each combination of
+-- accessor and constructor/mutator. So for the accessor @f@, and
+-- constructor/mutator @g@, we have one axiom of the form @f (g (...)) = ...@.
+--
+-- If however, this is cumbersome, we can often just test against an existing
+-- implementation. This is like a homomorphism test (but it's not quite a
+-- homomorphism). For unions, this is like saying if we have two lists A and B,
+-- and we take their union as lists, then sort and nub (remove duplicates)
+-- this is the same as using @Set.fromList@ to make them into sets, taking
+-- the union as sets, converting back with @Set.toList@ and then sorting
+-- the output list.
+--
+-- To have such a test replace an axiom, however, we need to have "homomorphisms"
+-- for each accessor. E.g., for all x and l, @member x (fromList l) = elem x l@.
```

Comment:
Like in #265: use `toList`, rather than `fromList`.

And probably phrase this in terms of semantics, rather than homomorphism.

You can spare yourself the cumbersomeness of using lists in your semantics if you use `Set` instead.

Reply:
I like homomorphisms better though :cry: It's like saying the semantics of operations on sets can be expressed in terms of the semantics of operations on lists because they can carry through the `fromList`. I'll add a comment stating this though.

Reply:
Also, I want to use lists in my semantics because we don't have a `fromSet` constructor and it won't correspond as well. (Though we easily could have such a constructor.)